doc/iterations/2023.6/shapes/filterTreeBasedRepresentations.adoc

```diff
@@ -0,0 +1,92 @@
+:imagesdir: images/
+= (M) Filtering tree based representations
+
+== Problem
+
+It is quite difficult to search for/identify specific tree item(s) in a tree when this tree contains lots of tree items.
+
+== Key Result
+
+The solution should not slow-down the user interface. The user interface must remains as smooth as it was before the developments of this filter.
+The results displayed by the filter should be visible immediately, i.e. without any latency.
+Otherwise, if the solution can only be implemented with a latency, then during the computation time, a "busy" indicator should be displayed, informing the user his request is processing.
+
+== Solution
+
+Add a search capacity in the Explorer View and other tree based representations, by providing a textual filtering of the tree just like what is being done in VSCode using a simple textfield.
+
+=== Breadboarding
+
+See the following breadboarding for the search:
+image:filterTreeBasedRepresentations-01.png[Filter widget]
+ 
+A shortcut (most likely ctrl+f) will be available in the Explorer View, allowing to display a search bar under the toolbar at the top of the Explorer View.
+This search bar will be composed by a simple textfield, a filter button and a close button.
+When user type a value in the textfield of the search bar, the tree item(s) containing this value are highlighted (only the part corresponding to the value, not the whole tree item(s)).
+Each character added/removed in the search bar will update the result (i.e. the highlighted tree item(s)).
+The filter button allows to hide all visible leaf tree items that do not match the search.
+The close button close the search bar, the highlighted results are not highlighted anymore, and the filtered tree items are not filtered anymore.
+The up/down arrow keys allow to jump from an highlighted item to the next one/previous one.
+
+=== Scenario 1 - Search
+
+User want to search tree items containing a specific value as their label or part of their label.
+User type the shortcut to enable the search bar.
+User type the value to search in the search bar.
+The tree items containing the searched value are highlighted (only the part of the items containing the value).
+Each character added/removed in the search bar updates the highlighted tree items.
+User close the search bar with the close button.
```

Comment:
As an end-user, I expect that:
- once I've found the item I searched for, hitting "Return" finishes the search (closes the search bar) and selects the element in question.
- if I don't find what I'm looking for, hitting "Esc" closes the search bar and returns the explorer to the previous state.

Reply:
I agree on Esc but since it's not searching, I don't see what hitting return should do when multiple tree items are matching the filter. Select them all? I don't see any behavior on return in VS Code

Reply:
Yes, the "Return" part of my comment was when I thought this was about an actual "search" and not "filtering".